src/lib/y2network/connection_config/vlan.rb

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2network/connection_config/base"
+
+module Y2Network
+  module ConnectionConfig
+    # Configuration for vlan connections
+    class Vlan < Base
+      # @return [Interface, nil]
+      attr_accessor :etherdevice
+      # @return [Integer, nil]
+      attr_accessor :vlan_id
+
+      # Replace the name of the etherdevice by a real Y2Network::Interface
+      #
+      # @see Y2Network::InterfacesCollection
+      #
+      # @param interfaces [Y2Network::InterfacesCollection]
+      def link_related_interfaces(interfaces)
+        return unless etherdevice
+        interface = interfaces.by_name(etherdevice)
+        self.etherdevice = interface if interface
+      end
+
+      def related_interfaces
+        [etherdevice].compact
```

Comment:
why do we need to return an array when vlan can be assigned to one device?

Reply:
It will be a common method for connections that could have related interfaces. Basically, only needed for reading and linked interfaces.